### PR TITLE
Added new Community List page to the login flow

### PR DESCRIPTION
**Pull request by JakeShirley**
Addresses #78

# Checklist
- [x] I have described what this PR contains

**Choose one of the following two options:**
- - [ ] This PR does not introduce major changes
- - [x] This PR introduces major changes, and I have consulted @buresdv, @jboi or @mormaer in the *Mlem Development Matrix room*

**Choose one of the following two options:**
- - [ ] This PR does not change the UI in any way
- - [x] This PR adds new UI elements / changes the UI, and I have attached pictures or videos of the new / changed UI elements

# Pull Request Information

## About this Pull Request
The PR add an Apollo-like communities screen.  It is injected between the "Accounts" screen and the "Community View".  This is a natural "home" screen that allows people to select what they want to browse when they first get into the app.

I am brand new to Swift and SwiftUI so please bear with me!

## Screenshots and Videos
image

Video demo: https://www.youtube.com/watch?v=R52NyvqnBug

## Additional Context
Still LOTS of clean up to do.  Still learning the ropes and seeing where I can make re-usable components and what not.

I wrong a bunch of (ugly) code to generate samples for the SwiftUI preview to function (yay!).  I rebased today (6/18) and had to disable a project setting for code coverage to get previews to work again (found the solution [here](https://stackoverflow.com/questions/59705079/swiftui-canvas-preview-compile-error-undefined-symbols-for-architecture-x86-64)).

### CODE REVIEW

File: Mlem/API/Requests/Community/ListCommunities.swift [hunk 1]
```diff
@@ -31,6 +31,7 @@ struct ListCommunitiesRequest: APIGetRequest {
 
             .init(name: "limit", value: limit.map(String.init)),
             .init(name: "page", value: page.map(String.init)),
+            .init(name: "type_", value: type.rawValue),
```
Comment (ShadowJonathan) [inline Mlem/API/Requests/Community/ListCommunities.swift:34]: 🤦 i somehow missed these...

thanks ^^

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Community List View.swift [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+            .frame(width: 8, height: 8)
+    }
+}
+
+///
+// Preview-only functions to populate the model
+// Not sure on what best practices are here
+// If these are useful these could be moved to be used externally
+///
+let fakeCommunityPrefixes: [String] =
```
Comment (ShadowJonathan) [inline Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Community List View.swift:354]: Maybe add a TODO here to move this to test code, or something of the like?

**Review by ShadowJonathan — COMMENTED** [review]
LGTM, however, when you press the "feed" button, does it now go to accounts or the list of communities to select and all?

Could we possibly have the default stack be `account -> list -> subscribed communities`?

**Comment by ShadowJonathan** [comment]
@buresdv why was this closed?

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Community List View.swift [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+        CommunityListView(
+            account: SavedAccount(id: 0, instanceLink: URL(string: "lemmy.com")!, accessToken: "abcdefg", username: "Test Account"),
+            testCommunities: fakeCommunityPrefixes.enumerated().map({
+                (index, element) in
+                generateFakeCommunity(id: index, namePrefix: element)
+            })
+        ).environmentObject(favoritesTracker)
+    }
+}
+
```
Comment (WestonHanners) [inline Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Community List View.swift:398]: Maybe we should move other components into separate files? I won't hold the PR up for this, it's just a thought.

**Comment by JakeShirley** [comment]
PR is ready! Moved some views out into their own files and cleaned up the UI a little bit.  Here it is:

image
image

**Comment by MrAusnadian** [comment]
Looks great @JakeShirley !

**Comment by buresdv** [comment]
I can't approve this in the current state. Will give more info when out of hospital

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Community List View/Community List View.swift [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+                                                // Filter down to sidebar entry which wants us
+                                                communitySection.sidebarEntry.contains(community: listedCommunity, isSubscribed: subscribedCommunities.contains(listedCommunity))
+                                            })
+                                    ) { listedCommunity in
+                                        CommuntiyFeedRowView(account: account, community: listedCommunity, subscribed: subscribedCommunities.contains(listedCommunity), communitySubscriptionChanged: self.hydrateCommunityData)
+                                    }
+                                }
+                            }
+                        
+                    }
```
Comment (EricBAndrews) [inline Mlem/Views/Tabs/Posts/Components/Post/Components/Community List View/Community List View.swift:80]: Not a blocker to this merge, but ideally the data filtering happens outside of the body--that enforces nice isolation of logic and rendering, and makes both parts more legible

**Review by EricBAndrews — APPROVED** [review]
Good to merge--I have left some stylistic/organizational notes, but nothing that can't wait until a follow-up PR.

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Community List View/Community List View.swift (hunk 4 above)
Comment (BrooklynMan) [reply Mlem/Views/Tabs/Posts/Components/Post/Components/Community List View/Community List View.swift:80]: if we can resolve conflicts on this, lets get it done and merge

**Review by BrooklynMan — APPROVED** [review]
well done

**Review by BrooklynMan — DISMISSED** [review]
LET'S GO ALREADYYYYYYY!

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Community List View/Community List View.swift (hunk 4 above)
Comment (JakeShirley) [reply Mlem/Views/Tabs/Posts/Components/Post/Components/Community List View/Community List View.swift]: Moved it to a utility method for now.  Definitely more cleanup to do.

**Review by BrooklynMan — APPROVED** [review]
well done

**Pull request merged**